### ISSUE TITLE
More programmatic control for virtual keyboards

### ISSUE THREAD

**Issue by shdwx-dev** [Yellow-Dog-Man/Resonite-Issues#2758]
### Is your feature request related to a problem? Please describe.

Since the userspace virtual keyboard can be freely modified and replaced, some users might want to implement more complex programmatic features such as quick cursor control (for example by sliding the cursor across the preview field), click-and-drag text selection, spell-check and error correction or even more complex features such as find-and-replace. When attempting this, it doesn't take long to realize that the VirtualKeyboard and related components are not very open and interactable, prompting workarounds or blocking implementations entirely.

### Describe the solution you'd like

- Ability to execute keypresses with flux while in userspace or a world in unsafe mode
- Ability to set cursor position by writing to a field or with a flux node that references the VirtualKeyboard component
- Ability to modify the string being edited directly using writes instead of using VirtualKeys with a driven AppendString (which can't delete characters and require a keypress)

### Describe alternatives you've considered

N/A

### Requesters

ShadowX

**Comment by shiftyscales** [comment]
> execute keypresses with flux while in userspace

#2585

Can you elaborate on this? What keys are you wanting to simulate/why?

> Ability to set cursor position

#1774

It is recommended to keep issues small / separate where possible. Compound issues like this one are less likely to be prioritized as the combination of all of those requests in the same issue make it a lot bigger than if they were separate requests, as noted above, two thirds of the solutions you proposed above already exist as separate issues.

**Comment by 5H4D0W-X** [comment]
> > execute keypresses with flux while in userspace
> 
> #2585
> 
> Can you elaborate on this? What keys are you wanting to simulate/why?

An example of how I might use that is a function to delete an entire word or sentence by detecting the nearest space or period character behind the caret. This is not currently possible as we can't execute an arbitrary number of delete/backspace events, only a fixed number by using a VirtualMultiKey (if that even works).

**Comment by Lixquid** [comment]
I don't believe https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2585 is a duplicate of the first bullet point of this issue (which is the specific thing I was about to make a feature request for); #2585 is asking for a new component that always forwards to the host machine (not within Resonite's input handling), whereas #2758 (this issue) is asking for a ProtoFlux component that just does the same thing as the already existing `VirtualKey` component.

To add a use-case of what I'd like it for; I currently have a keyboard that replaces the entire text of the input by doing a `Ctrl+A` then Append operation, which requires a secondary key event source as keys (even `VirtualMultiKey`s) can only send one key at a time. With a ProtoFlux component (or some other kind of direct write solution), I could just emit the keypress an update after the initial press has happened.
